### ISSUE TITLE
`Hide` is not working

### ISSUE THREAD

**Issue by MarvinJWendt** [charmbracelet/vhs#130]
I just noticed that `Hide` is not working on my machine using docker. The README looks like it too:

![image](https://user-images.githubusercontent.com/31022056/199369143-908246c3-c02d-4bf1-beee-51ae976b699f.png)

**Comment by maaslalani** [comment]
The hide command will hide frames from being recorded not the actual content of what is being typed. That is why the README shows that the command is already typed and you do not see the typing in the GIF but the typing does occur on the terminal.

**Comment by MarvinJWendt** [comment]
Oh okay, got it! Is there any way to hide the full setup? Using the docker env I have to install some packages as they are not already in alpine (like curl), but I'd like to make them invisible from the user.

**Comment by maaslalani** [comment]
Yep, do the Hide and all the typing and then clear the screen (by typing "clear" and  Enter)

**Comment by MarvinJWendt** [comment]
Okay this seems to (kinda) work! The first few frames are still showing the output of `apk add curl`, but I think this will eventually be handled by #39 ?

**Comment by maaslalani** [comment]
> Okay this seems to (kinda) work! The first few frames are still showing the output of `apk add curl`, but I think this will eventually be handled by #39 ?

Yes we'll definitely sort out the frames issue but to prevent it for now I think you can sleep for a little but after you've cleared the screen (but before the show)

**Comment by maaslalani** [comment]
But yes soon we'll have it so theres not extra frames cutting in.

**Comment by MarvinJWendt** [comment]
> > Okay this seems to (kinda) work! The first few frames are still showing the output of `apk add curl`, but I think this will eventually be handled by #39 ?
> 
> Yes we'll definitely sort out the frames issue but to prevent it for now I think you can sleep for a little but after you've cleared the screen (but before the show)

Works like a charm<sup>bracelet</sup> 😉

**Comment by MarvinJWendt** [comment]
Just out of interest, why does `Hide` not hide the full command? IMO it's not actually "hiding" the stuff between `Hide` and `Show`, and for me it's not very obvious that `Hide` just disables the typing animation. Maybe a flag like `Hide@animations` and/or `Hide@all` would do the trick?

**Comment by maaslalani** [comment]
Oh interesting, that is a very good idea and would be quite useful, IMO. I generally agree that it is a little confusing what Hide does and maybe needs a bit more documentation or a rename.

**Comment by griimick** [comment]
> Oh interesting, that is a very good idea and would be quite useful, IMO. I generally agree that it is a little confusing what Hide does and maybe needs a bit more documentation or a rename.
> 
> Maybe `Skip` is a better name? And its counter part is Resume/Continue/Play?

I too had trouble understanding it initially. `Pause`/`Resume` _recording_ sounds better to me.

**Comment by normanr** [comment]
Pause and Resume will probably be the least confusing. Interestingly they behave very similar to `Xoff`(^S) and `Xon`(^Q).

(Also for what it's worth: compressed video players have to do something similar when doing a seek to a frame between two keyframes: they have to pause the video decoder's output, send it the previous keyframe, followed by all frames up to the frame it wants to show, then re-enable output and only then send it the frame it wants to display)

**Comment by okineadev** [comment]
I solved the problem like this by running `clear` command before `Show`:

```diff
Hide
Type 'vhs_sandbox="$(mktemp -d)"' Enter
Type 'cd "$vhs_sandbox"' Enter
+ Type 'clear' Enter
Show

...
```

**Comment by mkonig** [comment]
Until there is a solution/decision it would be nice to at least have the readme extended with a hint that one needs to use clear so that the commands are not visible in the output.

I know there is an example in the readme using 'clear' and it is described that Hide stops capturing frames. But it still took me days to understand that Hide/Show do not hide what is typed.

To me it is a bit of a context switch that almost all other commands kind of operate on how the terminal is used. But Hide/Show operate on the capturing part.
